plugins/uefi-dbx/fu-uefi-dbx-device.c

```diff
@@ -110,18 +134,22 @@ fu_uefi_dbx_device_set_version_number(FuDevice *device, GError **error)
 		FuEfiSignature *sig = g_ptr_array_index(sigs, sigs->len - 1);
 		g_autofree gchar *csum =
 		    fu_firmware_get_checksum(FU_FIRMWARE(sig), G_CHECKSUM_SHA256, NULL);
-		if (csum != NULL)
-			fu_device_add_checksum(device, csum);
+		if (csum != NULL) {
+			if (!fu_uefi_dbx_device_set_checksum(self, csum, error))
+				return FALSE;
+		}
 	}
 
-	fu_device_set_version(device, fu_firmware_get_version(dbx));
+	/* success */
 	return TRUE;
 }
 
 static void
 fu_uefi_dbx_device_version_notify_cb(FuDevice *device, GParamSpec *pspec, gpointer user_data)
 {
 	fu_device_set_version_lowest(device, fu_device_get_version(device));
+	if (!fu_common_is_live_media())
+		fu_device_add_flag(device, FWUPD_DEVICE_FLAG_UPDATABLE);
```

Comment:
I don't know that this check makes sense anymore.  We already check everything on the identified ESP.  If someone boots live media why can't they run updates?

Reply:
I think the logic was that you're checking the LiveCD ESP, not the runtime ESP -- i.e. the LiveUSB disk might very well be fine with the new dbx, but your actual running system wouldn't be.

Reply:
Well I think a better way to approach that is to interrogate ALL. Partitions that can be an ESP? No?

Reply:
Yes, and after looking I think that's what we do now. I'll remove the livecd bit.

Reply:
Tear it out of dbxtool too.  And then at that point, do we even need the `fu_common_is_live_media`? :smile:

Reply:
https://github.com/fwupd/fwupd/pull/8353